java-checks/src/main/java/org/sonar/java/checks/ImmediateReverseBoxingCheck.java

```diff
@@ -157,13 +171,25 @@ private void checkForBoxing(ExpressionTree expression) {
     return null;
   }
 
-  private void addBoxingIssue(Tree tree, Symbol classSymbol, Tree boxingArg) {
+  private void addBoxingIssue(Tree tree, Symbol classSymbol, Tree boxingArg, Tree originalTree) {
+    String message;
     if (boxingArg.is(Tree.Kind.IDENTIFIER)) {
       IdentifierTree identifier = (IdentifierTree) boxingArg;
-      reportIssue(tree, "Remove the boxing of \"" + identifier.name() + "\".");
+      message = String.format("Remove the boxing of \"%s\".", identifier.name());
```

Comment:
you could just set the name and use the formatted directly on line 186.

Reply:
In fact, the message changes slightly between the two cases: `boxing of` vs `boxing to`.